sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3187,7 +3187,24 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
       ctx: PropertyListContext): Map[String, String] = withOrigin(ctx) {
     val properties = ctx.property.asScala.map { property =>
       val key = visitPropertyKey(property.key)
-      val value = visitPropertyValue(property.value)
+      // A property value can be String, Integer, Boolean or Decimal. Here we extract the property
+      // value based on whether its a string, integer, boolean or decimal literal.
+      val value =
+      if (property.value == null) {
+        null
+      } else {
+        expression(property.value) match {
+          case Literal(str: UTF8String, StringType) => str.toString
+          case Literal(_, BooleanType) => property.value.getText.toLowerCase(Locale.ROOT)
+          case Literal(_, IntegerType | DecimalType()) => property.value.getText
```

Comment:
Are there any reasons to support only literals but not foldable expressions?

Reply:
Yea we could support foldable expressions in the future. And constant-fold them, and then store the text of the resulting constant in the option value instead.

My idea was to separate concerns by doing just the parser change here first, but not doing this constant folding or changing the behavior yet. This PR only supports the same types of literal values for the option values as before the PR, simply improving error messages in other cases from generic `syntax error` to more descriptive messages indicating that only certain literals are supported. Then in another PR, we could implement the constant-folding.

But if you prefer, I can implement the constant-folding in this PR as well, please let me know.

Reply:
Seems no new syntax is supported after this PR, while the title says support general expressions... I prefer having the constant folding in this PR

Reply:
OK, this is done.